eng/terraform/main.tf

```diff
@@ -0,0 +1,119 @@
+provider "azurerm" {
+  features {}
+  #skip_provider_registration = "true"
+}
+
+locals {
+  full_prefix          = "${var.prefix}-${var.label}"
+  rg_name              = "${var.prefix}-${var.label}"
+  vnet_name            = "${local.full_prefix}-vnet"
+  short_label          = replace(var.label, "-", "")
+  long_sa_name         = "${local.short_label}tfstate${random_id.rand_storage.hex}"
+  storage_account_name = substr(local.long_sa_name, 0, 24)
+}
+
+# Create RG
+resource "azurerm_resource_group" "base_rg" {
+  name     = local.rg_name
+  location = var.location
+}
+
+# Create networking
+module "network" {
+  source = "./modules/network"
+
+  resource_group_name = azurerm_resource_group.base_rg.name
+  location            = azurerm_resource_group.base_rg.location
+  prefix              = local.full_prefix
+  vnet_name           = local.vnet_name
+  subnet_name         = var.base_subnet
+  vnet_cidr           = "10.1.0.0/16"
+  subnet_cidr         = "10.1.0.0/24"
+}
+
+# Database VM
+module "sql_vm" {
+  source              = "./modules/vm"
+  resource_group_name = azurerm_resource_group.base_rg.name
+  location            = azurerm_resource_group.base_rg.location
+  subnet_id           = module.network.vnet_subnet_id[0]
+  vm_sg_id            = module.network.vm_sg_id
+  prefix              = local.full_prefix
+
+  application        = "sql"
+  computer_name      = var.sql_vm_computer_name
+  vm_size            = var.sql_vm_size
+  data_disk_size     = var.sql_vm_data_disk_size
+  vm_image_publisher = var.sql_vm_image_publisher
+  vm_image_offer     = var.sql_vm_image_offer
+  vm_image_sku       = var.sql_vm_image_sku
+
+  admin_username = var.admin_username
+  admin_password = var.admin_password
+}
+
+# Web VM
+module "web_vm" {
+  source              = "./modules/vm"
+  resource_group_name = azurerm_resource_group.base_rg.name
+  location            = azurerm_resource_group.base_rg.location
+  subnet_id           = module.network.vnet_subnet_id[0]
+  vm_sg_id            = module.network.vm_sg_id
+  prefix              = local.full_prefix
+
+  application        = "web"
+  computer_name      = var.web_vm_computer_name
+  vm_size            = var.web_vm_size
+  data_disk_size     = var.web_vm_data_disk_size
+  vm_image_publisher = var.web_vm_image_publisher
+  vm_image_offer     = var.web_vm_image_offer
+  vm_image_sku       = var.web_vm_image_sku
+
+  admin_username = var.admin_username
+  admin_password = var.admin_password
+}
+
+# Test Runner VM
+
+module "runner_vm" {
+  source              = "./modules/vm"
+  resource_group_name = azurerm_resource_group.base_rg.name
+  location            = azurerm_resource_group.base_rg.location
+  subnet_id           = module.network.vnet_subnet_id[0]
+  vm_sg_id            = module.network.vm_sg_id
+  prefix              = local.full_prefix
+
+  application        = "runner"
+  computer_name      = var.runner_vm_computer_name
+  vm_size            = var.web_vm_size
```

Comment:
Could we have separate variables for test VM instead of using the web VM settings?

Reply:
added runner_vm_size variable